### PR TITLE
feat: side navigation expand icon

### PR DESCRIPTION
**Pull request by JKMarkowski**
## Related Issue
Closes https://github.com/SAP/fundamental-styles/issues/787
Closes https://github.com/SAP/fundamental-styles/issues/788
## Description
There is added new icon element, which replaced old arrow pseudo element. Icon is represented by `fd-nested-list__expand-icon` element.

Icon also has been aligned with fiori3 newest specs from side nav specs.

Group headers are removed from condensed mode. This rule has been added to description.

Breaking changes:
Old:
```html
<li class="fd-nested-list__item">
    <a class="fd-nested-list__link has-child">
        <span class="fd-nested-list__title">Level 1 Item</span>
    </a>
</li>
```

New:
```html
<li class="fd-nested-list__item">
    <div class="fd-nested-list__content has-child" tabindex="0">
        <a class="fd-nested-list__link" tabindex="-1">
            <span class="fd-nested-list__title">Level 1 Item</span>
        </a>
        <a class="fd-nested-list__expand-icon" tabindex="-1"></a>
    </div>
</li>
```

### Before:
image
image

### After:
image
image

### CODE REVIEW

**Comment by InnaAtanasova** [comment]
It's not related to the scope of this PR but maybe it's good idea to add some animation when we expand the second/third level of the Side Nav. Now it "jumps" and it's not the best UX behaviour. We can probably reach the designers for animation values @estelaxu3 @droshev

File: docs/pages/components/side-navigation.md [hunk 1]
```diff
@@ -373,13 +383,11 @@ The lists in both sections (Main and Utility) should have the `fd-nested-list--c
 
 ## Side navigation with icons - condensed state
 Use the `fd-side-nav--condensed` modifier class for `condensed` state.
+Group headers are forbidden in `condensed` state.
```
Comment (InnaAtanasova) [inline docs/pages/components/side-navigation.md]: I think we can replace this text with:
The following requirements must be met:
- all first level items of the Side Navigation should have icons
- the Side Navigation should not have Group Headers

@Vanessa-Cusmich

**Comment by estelaxu3** [comment]
I agree. We can propose this to designers.

File: docs/pages/components/side-navigation.md (hunk 1 above)
Comment (stefanoScalzo) [reply docs/pages/components/side-navigation.md:386]: agreed

Comment (Vanessa-Cusmich) [reply docs/pages/components/side-navigation.md:386]: Agreed +1

Comment (JKMarkowski) [reply docs/pages/components/side-navigation.md]: Descriptions are changed.

**Comment by droshev** [comment]
@JKMarkowski can you open an issue in NGX for bringing this change? Thanks!

**Pull request merged**